activesupport/test/xml_mini/common.rb

```diff
@@ -0,0 +1,229 @@
+    assert_equal_rexml(<<-eoxml)
+    <products/>
+    eoxml
+  end
+
+  def test_one_node_with_attributes_document_as_hash
+    assert_equal_rexml(<<-eoxml)
+    <products foo="bar"/>
+    eoxml
+  end
+
+  def test_products_node_with_book_node_as_hash
+    assert_equal_rexml(<<-eoxml)
+  <products>
+    <book name="awesome" id="12345" />
+  </products>
+  eoxml
+  end
+
+  def test_products_node_with_two_book_nodes_as_hash
+    assert_equal_rexml(<<-eoxml)
+  <products>
+    <book name="awesome" id="12345" />
+    <book name="america" id="67890" />
+  </products>
+  eoxml
+  end
+
+  def test_single_node_with_content_as_hash
+    assert_equal_rexml(<<-eoxml)
+    <products>
+      hello world
+    </products>
+  eoxml
+  end
+
+  def test_children_with_children
+    assert_equal_rexml(<<-eoxml)
+  <root>
+    <products>
+      <book name="america" id="67890" />
+    </products>
+  </root>
+  eoxml
+  end
+
+  def test_children_with_text
+    assert_equal_rexml(<<-eoxml)
+  <root>
+    <products>
+      hello everyone
+    </products>
+  </root>
+  eoxml
+  end
+
+  def test_children_with_non_adjacent_text
+    assert_equal_rexml(<<-eoxml)
+  <root>
+    good
+    <products>
+      hello everyone
+    </products>
+    morning
+  </root>
+  eoxml
+  end
+
+  def test_parse_from_io
+    jdom_skip
+
+    io = StringIO.new(<<-eoxml)
+  <root>
+    good
+    <products>
+      hello everyone
+    </products>
+    morning
+  </root>
+  eoxml
+    assert_equal_rexml(io)
+  end
+
+  def test_children_with_simple_cdata
+    jdom_skip
+
+    assert_equal_rexml(<<-eoxml)
+  <root>
+    <products>
+       <![CDATA[cdatablock]]>
+    </products>
+  </root>
+  eoxml
+  end
+
+  def test_children_with_multiple_cdata
+    jdom_skip
+
+    assert_equal_rexml(<<-eoxml)
+  <root>
+    <products>
+       <![CDATA[cdatablock1]]><![CDATA[cdatablock2]]>
+    </products>
+  </root>
+  eoxml
+  end
+
+  def test_children_with_text_and_cdata
+    jdom_skip
+
+    assert_equal_rexml(<<-eoxml)
+  <root>
+    <products>
+      hello <![CDATA[cdatablock]]>
+      morning
+    </products>
+  </root>
+  eoxml
+  end
+
+  def test_children_with_blank_text
+    jdom_skip
+
+    assert_equal_rexml(<<-eoxml)
+  <root>
+    <products>   </products>
+  </root>
+  eoxml
+  end
+
+  def test_children_with_blank_text_and_attribute
+    jdom_skip
+
+    assert_equal_rexml(<<-eoxml)
+  <root>
+    <products type="file">   </products>
+  </root>
+  eoxml
+  end
+
+  private
+    def adapter_name
+      raise NotImplementedError
+    end
+
+    def assert_equal_rexml(xml)
+      parsed_xml = ActiveSupport::XmlMini.parse(xml)
+      xml.rewind if xml.respond_to?(:rewind)
+      hash = ActiveSupport::XmlMini.with_backend("REXML") { ActiveSupport::XmlMini.parse(xml) }
+      assert_equal(hash, parsed_xml)
+    end
+
+    def expansion_attack_error
+      raise NotImplementedError
+    end
+
+    def jdom_skip
+      skip "JDOM does not support this test" if adapter_name == "JDOM"
+    end
```

Comment:
I'd think of the methods skipped on `JDOM` as extended engine tests — because for the common module or class to know about JDOM seems off — and would abstract like this:

```ruby
def test_skipped
  require_extended_engine

  assert_equal_rexml # yada yada...
end

private
  def require_extended_engine
    skip "#{engine} is just a basic engine" unless extended_engine?
  end

  def extended_engine? # Override in the JDOM tests and return false.
    true
  end
end
```

The `extended` term could stand to improve, same with the `skip` message.